azimuth/modules/model_performance/metrics.py

```diff
@@ -58,65 +64,75 @@ def first_value(di: Optional[Dict]) -> Optional[float]:
 class MetricsModule(FilterableModule[ModelContractConfig]):
     """Computes different metrics on each dataset split."""
 
-    def compute_on_dataset_split(self) -> List[MetricsModuleResponse]:  # type: ignore
-        ds: Dataset = assert_not_none(self.get_dataset_split())
-        indices = self.get_indices()
-        if len(indices) == 0:
+    def compute_metrics(self, ds: Dataset) -> List[MetricsModuleResponse]:
+        """Compute all metrics on a given dataset split.
```

Comment:
It took me some time (and separate conversation!) to understand why we needed to move this code into `compute_metrics` when it's still (aaaaaalmost) all that's being done in `compute_on_dataset_split()`. I think it would be useful to have a comment somewhere in the code documenting this rationale. Maybe the best way to do that would be to revise the docstring more or less like this? (But there could be a better way that I'm not thinking of.)
```suggestion
        """Compute all metrics on a given dataset subset (partial or full dataset split).
             Lives outside of `compute_on_dataset_split()` so that it can be called without module computation.
```
"subset" isn't necessarily right, but I wanted to indicate that it doesn't have to be a "dataset split," i.e. train or eval.

Reply:
If you think this is helpful, we could consider it for the bin one as well, but for some reason that didn't stand out to me in the same way - maybe for the simple reason that its `compute_on_dataset_split()` has a little more code? I'm not sure.